components/Chips/tests/unit/ChipViewRippleTests.m

```diff
@@ -0,0 +1,148 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import <XCTest/XCTest.h>
+
+#import "MaterialChips.h"
+#import "MaterialInk.h"
+#import "MaterialRipple.h"
+
+@interface MDCChipView (Testing)
+@property(nonatomic, strong) MDCStatefulRippleView *rippleView;
+@property(nonatomic, strong) MDCInkView *inkView;
+@end
+
+/**
+ This class confirms behavior of @c MDCChipView when used with @c MDCStatefulRippleView.
+ */
+@interface ChipViewRippleTests : XCTestCase
+
+@property(nonatomic, strong, nullable) MDCChipView *chipView;
+
+@end
+
+@implementation ChipViewRippleTests
+
+- (void)setUp {
+  [super setUp];
+
+  self.chipView = [[MDCChipView alloc] init];
+}
+
+- (void)tearDown {
+  self.chipView = nil;
+
+  [super tearDown];
+}
+
+/**
+ Test to confirm behavior of initializing a @c MDCChipView without any customization.
+ */
+- (void)testDefaultChipViewBehaviorWithRipple {
+  // Then
+  XCTAssertNotNil(self.chipView.rippleView);
+  XCTAssertEqualObjects([self.chipView.rippleView rippleColorForState:MDCRippleStateNormal],
+                        [UIColor colorWithWhite:0 alpha:(CGFloat)0.12]);
+  XCTAssertEqualObjects([self.chipView.rippleView rippleColorForState:MDCRippleStateHighlighted],
+                        [UIColor colorWithWhite:0 alpha:(CGFloat)0.12]);
+  XCTAssertEqual(self.chipView.rippleView.rippleStyle, MDCRippleStyleBounded);
+  XCTAssertFalse(self.chipView.enableRippleBehavior);
+  XCTAssertNil(self.chipView.rippleView.superview);
+  XCTAssertFalse(self.chipView.rippleView.allowsSelection);
+  CGRect chipViewBounds = CGRectStandardize(self.chipView.bounds);
+  CGRect rippleBounds = CGRectStandardize(self.chipView.rippleView.bounds);
+  XCTAssertTrue(CGRectEqualToRect(chipViewBounds, rippleBounds), @"%@ is not equal to %@",
+                NSStringFromCGRect(chipViewBounds), NSStringFromCGRect(rippleBounds));
+}
+
+/**
+ Test to confirm that setting @c enableRippleBehavior adds the @c rippleView as a subview.
+ */
+- (void)testEnableRippleBehaviorAddsRippleViewAsSubviewOfButton {
+  // When
+  self.chipView.enableRippleBehavior = YES;
+
+  // Then
+  XCTAssertNil(self.chipView.inkView.superview);
+  XCTAssertEqualObjects(self.chipView.rippleView.superview, self.chipView);
+}
+
+/**
+ Test to confirm toggling @c enableRippleBehavior removes the @c rippleView as a subview.
+ */
+- (void)testSetEnableRippleBehaviorToYesThenNoRemovesRippleViewAsSubviewOfButton {
+  // When
+  self.chipView.enableRippleBehavior = YES;
+  self.chipView.enableRippleBehavior = NO;
+
+  // Then
+  XCTAssertEqualObjects(self.chipView.inkView.superview, self.chipView);
+  XCTAssertNil(self.chipView.rippleView.superview);
+}
+
+/**
+ Test setting @c inkColor correctly sets the @c rippleColor on @c rippleView of the chip view.
+ */
+- (void)testSetCustomInkColorUpdatesRippleViewForHighlightedState {
+  // Given
+  UIColor *fakeColor = UIColor.redColor;
```

Comment:
Tip: Why is this a "fake color"? It looks like a normal one.

Reply:
Done.